src/Orbit/main.cpp

```diff
@@ -327,9 +327,20 @@ static void TryMoveSavedDataLocationIfNeeded() {
                    [&dialog, main_thread_id]() {
                      CHECK(main_thread_id == std::this_thread::get_id());
                      dialog.AddText("Finished.");
-                     dialog.EnableCloseButton();
+                     dialog.OnMoveFinished();
                    });
 
+  QObject::connect(&process, &orbit_qt::MoveFilesProcess::processInterrupted, &dialog,
+                   [&dialog, main_thread_id]() {
+                     CHECK(main_thread_id == std::this_thread::get_id());
+                     dialog.AddText("Interrupted.");
+                     dialog.OnMoveInterrupted();
+                   });
+
+  // Intentionally use the version of `QObject::connect` with three arguments.
+  QObject::connect(&dialog, &orbit_qt::MoveFilesDialog::interruptionRequested,
+                   [&process]() { process.RequestInterruption(); });
```

Comment:
I don't fully understand. Why not `QObject::connect(&dialog, &orbit_qt::MoveFilesDialog::interruptionRequested, &process, &orbit_qt::MoveFilesProcess::RequestInterruption);`?

Reply:
Because then it happens on the event loop (or whatever) of `MoveFilesProcess`, which is already executing the move, while I need to set `interruption_requested_` from another thread. Is there a better way?

Reply:
It might have something to do with `moveToThread(&background_thread_);`. That was from Dimitry's initial change so I'm not sure if that's needed.

Reply:
Ahh, now I see what is happening here. Thanks for the explanation. I think the code is fine.

The following would be more Qt-ish but also more work than just a quick fixup. I'm not even sure if it improves the situation:

1. Copying all the files happens in a single job on the event loop inside of the QThread in `MoveFilesProcess`.
2. QThread already has a mechanism to stop what it's currently doing. `QThread::quit`. It will quit the current event loop but the event loop will still finish executing the currently running task.
3. To make use of that you have to split `MoveFilesProcess::Run` into smaller tasks which all need to be queued on the event loop (with `QMetaObject::invokeMethod(..., Qt::QueuedConnection)` for example).
4. Then you could call `QThread::quit` from wherever you want since it's a thread-safe function. No atomic required etc.

Reply:
Thanks for the detailed explanation. I'm certainly leaning towards keeping it as it is as it seems to work and the new approach would need quite some more testing. But you understand what's going on here much better than me so you tell me whether you think it's worth investing the time.

Reply:
I think you should leave it as-is. It's not worth it, at least not at the moment.

Reply:
In the meantime I clarified the comment.

Reply:
Alright, thanks!